ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -186,31 +188,81 @@ public void shutdown (boolean async)
    }
 
     /**
-     * make all scanning operations suspend.
+     * <p>Make all scanning operations suspend.
+     * <p>This switches the recovery operation mode to <code>SUSPENDED</code>.
+     * Any attempt to start a new scan either by an ad hoc threads or by the periodic
+     * recovery thread will suspend its thread until the mode changes.
+     * If a scan is in progress when this method is called it will complete its scan
+     * without suspending.
+     * <p>Note that this method is also influenced by
+     * {@link com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean#setWaitForRecovery}.
+     * In case <code>{@link RecoveryEnvironmentBean#isWaitForRecovery()} == true</code>,
+     * it is <b>very important</b> that, before invoking this method, the following
+     * pre-conditions are met:
+     * * The Transaction System gets disabled (i.e. {@link TxControl#disable()})
+     * (i.e. no new transactions will be created)
+     * * {@link com.arjuna.ats.arjuna.coordinator.TransactionReaper} finishes monitoring all in-flight
+     * transactions
+     * * All transactions without a timeout are beyond the prepare phase of the 2PC protocol;
+     * to check when this condition is verified, {@link com.arjuna.ats.arjuna.coordinator.ActionManager}
+     * can be employed
      *
-     * This switches the recovery operation mode to SUSPENDED. Any attempt to start a new scan either by an ad hoc
-     * threads or by the periodic recovery thread will suspend its thread until the mode changes. If a scan is in
-     * progress when this method is called it will complete its scan without suspending.
-     *
-     * @param async false if the calling thread should wait for any in-progress scan to complete before returning
-     * @return the previous mode before attempting the suspend
+     * @param async false if the calling thread should wait for any in-progress scan to complete before returning.
+     * In case <code>{@link RecoveryEnvironmentBean#isWaitForRecovery()} == true</code>, this parameter is
+     * override.
+     * @return the previous mode before attempting the suspension
      */
-
    public Mode suspendScan (boolean async)
    {
        synchronized (_stateLock)
        {
-           // only switch and kick everyone if we are currently ENABLED
            Mode currentMode = getMode();
 
+           /*
+            * isWaitForRecovery checks whether suspension should delay until RecoveryModule
+            * implementations recover all their transactions.
+            */
+           if (currentMode == Mode.ENABLED && recoveryPropertyManager.getRecoveryEnvironmentBean().isWaitForRecovery()) {
+
+               /*
+                * At least, one clear scan (after the Transaction Manager has been shut down)
```

Comment:
I meant, how does the code validate that the TM has been shutdown/disabled. But you explained in person that the integrator has to do that - but rather than commenting the code should either check the condition or throw an exception if the TM is still running.

Reply:
After re-reading the comment, I can accept that it is informational only and is not an assertion. Therefore I will resolve this conversation.